### PR TITLE
Feature/179 script parsed event

### LINKED ISSUE #172: Add script for making data-init backward compatible
As a developer I want migration from v1 or v2 to v3 as easy as possible. This includes also that I get the script that makes data-init backward compatible either in the docs or as a script I can import.

### LINKED ISSUE #179: resourcesReady does not mean a script has finished parsing
The `resourcesReady` event does not take into account the fact that scripts might not yet have finished loading/being parse.
This creates a racing condition on certain projects, especially ones that attach component contructor functions to `jQuery`.

Screenshot 2020-05-28 at 16 16 38

### PR DESCRIPTION
**Pull request by ctmm**
Closes #179 
Closes #172

### CODE REVIEW

**Comment by timomayer** [comment]
Thanks for this PR @ctmm i really like what scriptParsedEvent does, not having this caused some trouble in the past. my thoughts are if we really need both events resourceReady and scriptParsedEvent. I do not see a scenario where resourceReady is really useful infact it is used like a broken scriptParsedEvent and we are using it all time with "fingers crossed" that script is already parsed when we fire this event.
So for simplcity reason and also improve older projects without changing the event i suggest to merge this events. it should work like scriptParsedEvent is working but should be named resourceReady (for backward compatability) what do you guys think?

**Comment by ctmm** [comment]
@timomayer
I kept it to empower devs, as the `resourcesReady` event might be useful for anyone wishing to perform some operation when styles or html are loaded. 

Only javascript resources  trigger the `scriptParsed` event.
 
I can see your concern about the event name, as it's historical use might confuse some devs, however:

1. I tried to make the docs as explicit as possible about the what each event means and does;
2. The legacy support/migration guide requires that you update a ~10 lines of code to make full use of the "initialize as soon as ready" feature of this new version. Under that light I considered that changing 1 extra event name wouldn't be additional effort;

For these reasons I would keep both.
I do not discard the option of renaming the `resourcesReady` event to something else, though.
 
Could you please give it a try and test this integration on an older project? Given your insight of our projects, this would give me confidence that the name changes won't be a hurdle for our teams migrating every project.

**Comment by timomayer** [comment]
Thanks for your answer Cesar,
I still think this is hard to understand for developers which are not deep into the topic. I guess most of them will still make the failure that they use resourceReady not only for css but also for js files and then run into problems which are hard to understand and debug.

I would still vote for combining it all in the resourceReady event to make it as easy as possible for devs to use it, but make this event as smart as possible (use logic you implemented for scriptsParsed for js files, use logic for css and html files like before)

I do think this simplification, easyness to use is worth not having the possibility having both events (scriptsParsed, resourceReady)

**Comment by SheepFromHeaven** [comment]
@timomayer @ctmm I do not agree with you @timomayer if I understood you right (You both keep using `resourceReady`and `resourcesReady` - note the plural in there - interchangebly. 
My opinion: Have 2 events, however they are called: 
1. Everything is done
2. One thing is done

As far as I understood you, you would want to drop feature 2 for ease of unexperienced developers. I do strongly disagree with this mindset. If this is not the case please clarify.

**Comment by timomayer** [comment]
no @SheepFromHeaven ;) think you got me and cesar wrong ;) think a call is more efficient. you me and cesar 15min?

**Comment by ctmm** [comment]
@timomayer just stepped out of call with Marc where we agreed upon how to refine this PR:

1. 
`resourcesReady` will be renamed to `resourceQueueEmpty`. 
This is because what's actually going happening when this event is fired is that the resource queue has been emptied. It fires everytime the contents of a `data-resources` have all been loaded, but not attached to the dom.

2.
`scriptParsed` stay as is.

3.
Two new events, named `styleAppended` and `htmlAppended` are fired from within the respective handlers as soon as a given resource has been appended.

4.
We do not implement any event that informs the app about everything being finished, because we don't currently have any logic that can support that feature.

Please let us know your thoughts!

**Comment by timomayer** [comment]
will you update the docs?

**Comment by ctmm** [comment]
I would like to have updated the docs before we merge this PR

File: docs/api.md [hunk 1]
```diff
@@ -6,17 +6,20 @@ sidebar_label: API guide
...
-| `baseMap`                        | [BaseMap](#basemap)   | yes      | undefined          | The base path map to use for resolving base references                      |
-| `handler`                        | [Handler](#handler)[] | yes      | `[]`               | These handlers will be called on resource loaded when they match            |
+| Property                         | Type                  | Optional | Default                | Description                                                                |
+|----------------------------------|-----------------------|----------|------------------------|--------------------------------------------------------------------------- |
+| `container`                      | HTMLElement           | yes      | body                   | The container in which to search for resource components                   |
+| `resourceListAtrributeSelector`  | string                | yes      | `data-resources`       | The attribute to search in container for the resources                     |
```
Comment (SheepFromHeaven) [inline docs/api.md:12]: Typo 'attribute'

File: docs/api.md [hunk 2]
```diff
@@ -76,3 +81,42 @@ const definition = {
     base: '/public/resources'
 }
 ```
+
+## Available events
+To communicate to your app, the resource loader will fire events based on the state of resources.
+These events can be renamed to your fit your needs, while retaining the same logic.
```
Comment (SheepFromHeaven) [inline docs/api.md:87]: 'To fit your needs...'

File: docs/initialize.md [hunk 3]
```diff
@@ -53,14 +54,38 @@ registerScript(
...
-    initScripts: false
+    initScripts: false,
 });
-```
\ No newline at end of file
+```
+
+2. If your scripts are self-initializing, you're done. If they register a contructor class, proceed with step 3.
+
+3. Add a listener to the `document`, which triggers your custom initialization logic:
```
Comment (SheepFromHeaven) [inline docs/initialize.md:75]: In your example you use window instead of document

**Pull request merged**